### DISCUSSION TITLE
Get Iterator index of previous loop

### DISCUSSION THREAD

**Discussion by koneubauer** [Budibase/budibase#11298] [Help] [answered]
I created an Automation in which my first step is executing a database-query which returns start and destination coordinates and other relevant data per row.
The next step is querying a Routing-Api which returns the distance between start and destination from step 1. On this step, i have enabled a loop which iterates over the rows of the first databse-query.
The third step is a create-row element with a loop which should insert the distance with the other data from the first step (id, name, date,...) into a table.
I can access the results of the Routing-Api query and i can also access the result-set from step 1. However, i would need an iterator index for getting the correlating data from step 1 to match the calculated distance of step 2
Is there any way to get the index of the loop of step 2 and use it in step 3 to get the matching data from step 1?
I have highlighted the value which would have to be the iterator index for it to work. (step_3)

![step_2](https://github.com/Budibase/budibase/assets/117980610/ec183462-29d3-4438-b3e7-a21d71af3397)
![step_3](https://github.com/Budibase/budibase/assets/117980610/e27862e6-5b11-4958-9e3b-07cf09554031)

**Comment by ConorWebb96** [comment, marked as answer]
Hey @koneubauer,

I would suggest that you look at the `JS SCRIPTING` action that you can add, with this you may be able to do the more complicated logic that you're after related to steps 1 and 2 you talked about. As I don't have your configuration I'd suggest doing something along these lines for the additional step after 1 and 2;
<img width="640" alt="Screenshot 2023-07-21 at 09 01 58" src="https://github.com/Budibase/budibase/assets/126772285/154434d8-cba4-4911-bffb-cd997fb3547c">
<img width="489" alt="Screenshot 2023-07-21 at 09 02 10" src="https://github.com/Budibase/budibase/assets/126772285/5a1a0db9-d13f-4b15-a63e-df953ee64647">

```
  const dataStep1 = {{ steps.1.response }};

  const resultsStep2 = [];
  for (let index = 0; index < dataStep1.length; index++) {
    const row = dataStep1[index];
    const start = row.start;
    const destination = row.destination;
    const distance = {{ steps.2.items }};
    const result = { distance: distance, otherData: row.otherData };
    resultsStep2.push(result); // output the combined data related to one another.
  }
```

From the above you could then create a row with all the relevant data. For the above you likely will need to change it to meet your specific needs. 

I hope this at least points you in the right direction!

**Comment by koneubauer** [comment]
Thanks a lot!
